spec/controllers/users/two_factor_authentication_controller_spec.rb

```diff
@@ -222,6 +223,55 @@ def index
       end
     end
 
+    context 'phone is not confirmed' do
+      before do
+        @user = create(:user)
+        @unconfirmed_phone = '+1 (202) 555-1213'
+
+        sign_in_before_2fa(@user)
+        subject.user_session[:context] = 'confirmation'
+        subject.user_session[:unconfirmed_phone] = @unconfirmed_phone
+      end
+
+      it 'sends OTP inline when confirming phone' do
+        allow(SmsOtpSenderJob).to receive(:perform_now)
+        get :send_code, params: { otp_delivery_selection_form: { otp_delivery_preference: 'sms' } }
+
+        expect(SmsOtpSenderJob).to have_received(:perform_now).with(
```

Comment:
I would not duplicate the `perform_now` expectation like this, let's just move this up top and combine/replace `allow()`...


```
expect(SmsOtpSenderJob).to receive(:perform_now).with(
  ...
)
```

Reply:
@zachmargolis so if I move it above, it sets the expectation to the current otp code, which changes on 238 when it `get`s a new one.